Gh-210: add doc for gaffer docker (#371)
gchq/gaffer-doc: docs/dev/kubernetes-guide/add-libraries.md

# Adding your own libraries and functions

By default with the Gaffer deployment you get access to the:

- Sketches library
- Time library
- Bitmap Library
- JCS cache library

If you want more libraries than this (either one of ours of one of your own) you will need to customise the docker images and use them in place of the defaults.

> You will need a basic Gaffer instance deployed on Kubernetes, [here](deploy-empty-graph.md) is how you do that.

Reviewer: It's better if link's are in context.
```suggestion
You will first need [a basic Gaffer instance deployed on Kubernetes](deploy-empty-graph.md).
```

> ## Overwrite the REST war file

Reviewer 2: ```suggestion
## Add Extra Libraries to Gaffer REST
```
The titles not really what the section is about it uses spring boot not webstart so no WAR files, and this is just about adding some custom JARs not overwriting the main one

At the moment, Gaffer uses a runnable jar file located at `/gaffer/jars`. When it runs it includes the `/gaffer/jars/lib` on the classpath. There is nothing in there by default because all the dependencies are bundled in to the JAR. However, if you wanted to add your own jars, you can do it like this:

```Dockerfile
FROM gchq/gaffer-rest:latest
COPY ./my-custom-lib:1.0-SNAPSHOT.jar /gaffer/jars/lib/
```

Build the image using:

```bash
docker build -t custom-rest:latest .
```

## Add the extra libraries to the Accumulo image

Gaffer's Accumulo image includes support for the following Gaffer libraries:

- The Bitmap Library
- The Sketches Library
- The Time Library

In order to push down any extra value objects and filters to Accumulo that are not in those libraries, we have to add the jars to the accumulo `/lib/ext directory`. Here is an example `Dockerfile`:

```Dockerfile
FROM gchq/gaffer:latest
COPY ./my-library-1.0-SNAPSHOT.jar /opt/accumulo/lib/ext
```

Then build the image

```bash
docker build -t custom-gaffer-accumulo:latest .
```

# Switch the images in the deployment

> You will need a way of making the custom images visible to the kubernetes cluster. With EKS, you can do this by uploading the images to ECR. There is an example for how to do that in one of our [other guides](aws-eks-deployment.md). With KinD, you just run `kind load docker-image <image:tag>`.

Reviewer: Remove the EKS bit if removing the EKS page.
